src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/Runtime/General/ThunkedApis.cs

```diff
@@ -199,7 +199,7 @@ public sealed override Type[] GetGenericArguments()
                 if (ns != null && !ns.Equals(ifc.Namespace))
                     continue;
                 if (match != null)
-                    throw new AmbiguousMatchException();
+                    ThrowHelper.ThrowAmbiguousMatchException(match);
```

Comment:
Again, when we hit this line, `match` is the _first_ one we found, and we're going to throw with it's info